reactive/kubernetes_master.py

```diff
@@ -2575,3 +2576,77 @@ def add_systemd_iptables_patch():
 
     # enable and run the service
     service_resume(service_name)
+
+
+@when('leadership.is_leader',
+      'kubernetes-master.components.started',
+      'endpoint.prometheus.joined',
+      'certificates.ca.available')
+def register_prometheus_jobs():
+    prometheus = endpoint_from_flag('endpoint.prometheus.joined')
+    # tls = endpoint_from_flag('certificates.ca.available')
+    loadbalancer = endpoint_from_flag('loadbalancer.available')
+    forced_lb_ips = hookenv.config('loadbalancer-ips').split()
+    hacluster_vip = get_hacluster_ip_or_hostname()
+
+    if forced_lb_ips:
+        # if the user gave us IPs for the load balancer, assume they know
+        # what they are talking about and use that instead of our information.
+        round_robin = get_unit_number() % len(forced_lb_ips)
+        k8s_api_endpoint = forced_lb_ips[round_robin]
+    elif hacluster_vip:
+        k8s_api_endpoint = hacluster_vip
+    elif loadbalancer:
+        lb_addresses = loadbalancer.get_addresses_ports()
+        k8s_api_endpoint = lb_addresses[0].get('public-address')
+    else:
+        k8s_api_endpoint = prometheus.ingress_address
+    k8s_password = get_password('basic_auth.csv', 'admin')
+
+    templates_dir = Path('templates')
+    for job_file in Path('templates/prometheus').glob('*.yaml.j2'):
+        prometheus.register_job(
```

Comment:
Is it cool to register the jobs every hook run? Should we use a flag to only do this once?

Reply:
We do want to re-check the values, in case they change, and if they don't then the rendered template should be stable so it shouldn't cause any extra hooks to fire on the other end, but I can add explicit logic to watch for changes in the input values if you think that's cleaner (explicit over implicit).